modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -3010,7 +3010,9 @@ cargo::optional<Error> Builder::create<OpPtrAccessChain>(
       basePointeeTy, base, indexes, module.getName(op->IdResult()),
       IRBuilder.GetInsertBlock());
 
-  if (basePointeeTy->isStructTy()) {
+  // Check member decorations if the original indexes field is not empty
+  // as in this case we are accessing an element
+  if (basePointeeTy->isStructTy() && indexes.size() > 1) {
```

Comment:
Maybe I'm missing something to do with the meaning of "original", but this condition isn't "not empty" - it's "has more than one index".

I'd maybe recommend `!indexes.empty()`. `empty` is guaranteed to be constant time operation, but `size` does not have that requirement.

Reply:
Oh sorry I believe as of C++11, `size` has that constant-time requirement too.

Reply:
I can't use empty() as the above code adds the element in, so it will always be at least one. I'll rephrase to make it clearer